tests/utils.py

```diff
@@ -55,27 +56,23 @@ def clone(data: NdarrayTensor) -> NdarrayTensor:
     return copy.deepcopy(data)
 
 
-def allclose(a: NdarrayTensor, b: NdarrayTensor) -> bool:
+def allclose(a: NdarrayOrTensor, b: NdarrayOrTensor, rtol=1.0e-5, atol=1.0e-8, equal_nan=False) -> bool:
     """
     Check if all values of two data objects are close.
 
     Note:
         This method also checks that both data objects are either Pytorch Tensors or numpy arrays.
 
     Args:
-        a (NdarrayTensor): Pytorch Tensor or numpy array for comparison
-        b (NdarrayTensor): Pytorch Tensor or numpy array to compare against
+        a (NdarrayOrTensor): Pytorch Tensor or numpy array for comparison
+        b (NdarrayOrTensor): Pytorch Tensor or numpy array to compare against
 
     Returns:
         bool: If both data objects are close.
     """
-    if isinstance(a, np.ndarray) and isinstance(b, np.ndarray):
-        return np.allclose(a, b)
-
-    if isinstance(a, torch.Tensor) and isinstance(b, torch.Tensor):
-        return torch.allclose(a, b)
-
-    return False
+    a = a.cpu() if isinstance(a, torch.Tensor) else a
+    b = b.cpu() if isinstance(b, torch.Tensor) else b
+    return np.allclose(a, b, rtol, atol, equal_nan)
```

Comment:
`np.testing.assert_allclose()` will print more info:
```python
>>> np.allclose([1e10,1e-7], [1.00001e10,1e-8])
False
>>> np.testing.assert_allclose([1e10,1e-7], [1.00001e10,1e-8])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/anaconda3/envs/py37/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 1531, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/usr/local/anaconda3/envs/py37/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 844, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 2 / 2 (100%)
Max absolute difference: 100000.
Max relative difference: 9.
 x: array([1.e+10, 1.e-07])
 y: array([1.00001e+10, 1.00000e-08])
```

Reply:
Thanks, updated to use `np.testing.assert_allclose`.